src/main/java/HeadersRestTemplateService.java

```diff
@@ -0,0 +1,28 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+@Service
+public class HeadersRestTemplateService {
+
+    @Value("${github.token}")
+    private String githubToken;
+
+    @Value("${openai.api.key}")
+    private String openaiApiKey;
+
+    public HttpHeaders createGitHubHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + githubToken);
+        headers.set("Content-Type", "application/json");
+        return headers;
+    }
```

Comment:
The method is straightforward but could benefit from a more generic approach to avoid code duplication with createOpenAiHeaders(). (Severity: medium)
**Suggestion**: Consider creating a private method that accepts parameters for the token and service type to reduce redundancy.